File: HotDesk_PRAI/Library/Library.Root/TemplateField/viewfield.vb
Imports System.Web.UI.WebControls

Public Class viewfield
    Implements System.Web.UI.ITemplate

    Public Const LabelHeaderID As String = "lblview"
    Public Const LiteralItemID As String = "ltrviewitem"
    Private plittype As ListItemType

    Public Sub New(ByVal type As ListItemType)
        MyBase.New()
        Me.plittype = type
    End Sub

    Public Sub InstantiateIn(ByVal container As System.Web.UI.Control) Implements System.Web.UI.ITemplate.InstantiateIn
        Dim ltritem As Literal
        Dim lbdel As Label

        Select Case Me.plittype
            Case listItemType.Header
                lbdel = New Label()
                lbdel.Text = "View"
                lbdel.ID = "lblview"
                lbdel.EnableViewState = False
                container.Controls.Add(lbdel)
            Case listItemType.Item
                ltritem = New Literal()
                ltritem.ID = "ltrviewitem"
                ltritem.EnableViewState = False
                container.Controls.Add(ltritem)
        End Select
    End Sub
End Class

File: HotDesk_PRAI/Library/Library.Root/TemplateField/viewfield.cs
using System.Web.UI.WebControls;

public class viewfield : System.Web.UI.ITemplate
{
    public const string LabelHeaderID = "lblview";
    public const string LiteralItemID = "ltrviewitem";
    private ListItemType plittype;

    public viewfield(ListItemType type)
    {
        this.plittype = type;
    }

    public void InstantiateIn(System.Web.UI.Control container)
    {
        Literal ltritem;
        Label lbdel;

        switch (this.plittype)
        {
            case ListItemType.Header:
                lbdel = new Label();
                lbdel.Text = "View";
                lbdel.ID = "lblview";
                lbdel.EnableViewState = false;
                container.Controls.Add(lbdel);
                break;
            case ListItemType.Item:
                ltritem = new Literal();
                ltritem.ID = "ltrviewitem";
                ltritem.EnableViewState = false;
                container.Controls.Add(ltritem);
                break;
        }
    }
}
